crates/shared/src/sources/uniswap_v3/pool_fetching.rs

```diff
@@ -169,7 +192,7 @@ impl PoolsCheckpointHandler {
             .get_pools_with_ticks_by_ids(&pool_ids, registered_pools.fetched_block_number)
             .await?
             .into_iter()
-            .filter_map(|pool| Some((pool.id, pool.try_into().ok()?)))
+            .filter_map(|pool| Some((pool.id, PoolInfo::try_from(pool).ok()?.into())))
```

Comment:
Why not to implement `TryFrom` for `PoolInfoHandler` directly instead of doing two conversions? same applies to [L269](https://github.com/cowprotocol/services/pull/2629/files#diff-8a0be017c847413b5a422604316ed606c59b2294e551106ea4aea93c95bde758R269).